src/aiida_quantumespresso/calculations/xspectra.py

```diff
@@ -108,4 +127,21 @@ def prepare_for_submission(self, folder):
         core_file_info = (core_file.uuid, core_file.filename, core_file.filename)
         calcinfo.local_copy_list.append(core_file_info)
 
+        # if included as an input, append the gamma file node to the copy list
+        if 'gamma_file' in self.inputs.keys():
+            gamma_file = self.inputs.gamma_file
+            gamma_file_info = (gamma_file.uuid, gamma_file.filename, gamma_file.filename)
+            calcinfo.local_copy_list.append(gamma_file_info)
+
+        # Check if the parent folder is an XspectraCalculation type, if so we then copy the
+        # save file to enable calculation restarts and re-plots.
+        parent_folder = self.inputs.parent_folder
+        parent_calcs = parent_folder.get_incoming(node_class=orm.CalcJobNode).all()
+        parent_calc = parent_calcs[0].node
+        restart_flag = parent_calc.process_type == 'aiida.calculations:quantumespresso.xspectra'
+        if restart_flag:
```

Comment:
Here is a useful shortcut
```suggestion
        parent_folder = self.inputs.parent_folder
        parent_calc = parent_folder.creator

        if parent_calc.process_type == 'aiida.calculations:quantumespresso.xspectra':
```